### ISSUE TITLE
[HOLD for @0xmiroslav payment] [$1000] mweb: Green background colour visible when scrolling on the Preference Page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#23422]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1 Open app
2 Go to Settings > Preference
3 Make a scroll
## Expected Result: 
When scrolling the colour remains Blue
## Actual Result: 
Some parts of the screen remain light green.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.44-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/20463c39-0137-46d1-aa12-37277dee6826

https://github.com/Expensify/App/assets/43996225/12dfc816-f6e0-4be9-9343-fa7c16a80e5d

**Expensify/Expensify Issue URL:**
**Issue reported by:** @narzullome
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690019575694929

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0135c9358446287985</li>
        <li>Upwork Job ID: 1683858764520570880</li>
        <li>Last Price Increase: 2023-09-05</li>
<li>Automatic offers: </li>
<ul>
<li>0xmiroslav | Reviewer | 26515161</li>
<li>ginsuma | Contributor | 26515162</li>
<li>narzullome | Reporter | 26515167</li>
<li>ishpaul777 | Contributor | 26515517</li>
</ul></li>
    </ul>
</details>

**Comment by ahmedGaber93** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
mweb: Green background colour visible when scrolling on the Preference Page
### What is the root cause of that problem?
the background color `themeColors.appBG` is set for whole view above scrollview not for the view wrap children
https://github.com/Expensify/App/blob/cd851d0fbd901c70a32220472104ffc887639f2a/src/components/IllustratedHeaderPageLayout.js#L42
### What changes do you think we should make in order to solve the problem?
remove background color `themeColors.appBG` from here
https://github.com/Expensify/App/blob/cd851d0fbd901c70a32220472104ffc887639f2a/src/components/IllustratedHeaderPageLayout.js#L42
and add it here  `, StyleUtils.getBackgroundColorStyle(themeColors.appBG)` for view wrap children here
https://github.com/Expensify/App/blob/cd851d0fbd901c70a32220472104ffc887639f2a/src/components/IllustratedHeaderPageLayout.js#L56


### What alternative solutions did you explore? (Optional)

**Comment by slafortune** [comment]
Bug report looks good!

**Comment by slafortune** [comment]
I added their name to the proposal title so it's easy to recognize who the proposal is from - do you think there is an easier way to show that we should have done?

**Comment by slafortune** [comment]
@0xmiroslav - when you get a chance, can you review @ginsuma's proposal?

**Comment by slafortune** [comment]
Hi @0xmiroslav  - this seems as it would be a straight forward fix, can you review this proposal?

**Comment by narzullome** [comment]
@slafortune  HI the same issue on the Workspace screen [](https://expensify.slack.com/archives/D05A6STJLUS/p1691305342xxxx12
)

**Comment by aditygrg2** [comment]
#Proposal

Please re-state the problem that we are trying to solve in this issue.

Non-matching background color on `IllustratedHeaderPageLayout` pages

##What is the root cause of that problem?

The present view component is not placed correctly. It should be placed above the ScrollView.

##What changes do you think we should make in order to solve the problem?

1. The `view` should be made one level above [here](https://github.com/Expensify/App/blob/b7d17e0fd33b83e9fbd45146b6123b6323ece79e/src/components/IllustratedHeaderPageLayout.js#L47). This is incorrectly set presently.
2. We can add background colors to the children's content of the page as well as the `FixedFooter` component
3. The top can be made as 0 and zIndex as a negative constant because the view should be present behind the ScrollView.

`overscrollSpacer: (backgroundColor, height) => ({
        backgroundColor,
        height,
        width: '100%',
        position: 'absolute',
        top: 0,
        zIndex: -height,
        left: 0,
        right: 0,
    }),`

After these changes, the views will look like this:

https://github.com/Expensify/App/assets/98523623/cd3ca396-3383-42d6-92fb-92dcea766862

The android and chrome are not affected by this because iOS and Safari native views only support the bouncing effect on overscroll.

**Comment by ishpaul777** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
on Safari - Green background colour visible when scrolling on the Preference Page and 

### What is the root cause of that problem?
The overscroller background overlay is not placed properly

### What changes do you think we should make in order to solve the problem?
we need to put the overscrollSpacer out of ScrollView component and modify the styles.
```jsx
<View style={[styles.flex1,]}>
    <View style={styles.overscrollSpacer(backgroundColor, windowHeight)}>
        <View style={[styles.flex1, StyleUtils.getBackgroundColorStyle(backgroundColor)]}/>
        <View style={[styles.flex1, StyleUtils.getBackgroundColorStyle(themeColors.appBG)]}/>
    </View>
    <ScrollView
        contentContainerStyle={safeAreaPaddingBottomStyle}
        showsVerticalScrollIndicator={false}
    >
        <View style={[styles.alignItemsCenter, styles.justifyContentEnd, StyleUtils.getBackgroundColorStyle(backgroundColor)]}>
            <Lottie
                source={illustration}
                style={styles.w100}
                autoPlay
                loop
            />
        </View>
        <View style={[styles.pt5, StyleUtils.getBackgroundColorStyle(themeColors.appBG)]}>{children}</View>
    </ScrollView>
    {!_.isNull(footer) && <FixedFooter style={[StyleUtils.getBackgroundColorStyle(themeColors.appBG)]}>{footer}</FixedFooter>}
</View>
```

and we need to modify overscrollSpacer flexbox with direction column. and make sure it is below the actual content by setting appropriate index value.

I will add results soon...

**Comment by slafortune** [comment]
@0xmiroslav do you have thoughts on these proposals?

**Comment by slafortune** [comment]
@0xmiroslav Yes, I agree.  I hadn't realized this was iOS natural behavior.

**Comment by roryabraham** [comment]
Yeah, I was aware of this and tried to fix it at the time, but ultimately I thought that it was a safari bug – the color of the native overscroll bounce area is supposed to be controlled by `meta[name=theme-color]` attribute, but found an iOS Safari bug where that attribute updates the status bar color but not the overscroll color.

I later discovered the same bug on iOS native and that's when I implemented the overscroll spacer to fix it.

Anyways, if the solution is as simple as moving the overscroll spacer up out of the ScrollView, then I'm all for it 👍🏼

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by slafortune** [comment]
@0xmiroslav are you able to reproduce this, or shall we close it?

**Comment by aditygrg2** [comment]
@slafortune It is reproducible on iOS, Safari.

**Comment by slafortune** [comment]
I'm not able to reproduce at all - On iOS app I can pull down and the blue stays nice - in iOS Safari it doesn't pull down at all and leaves no space

https://github.com/Expensify/App/assets/11725910/b2c1b00c-ab55-4eb2-a215-23b09afb31e7

**Comment by slafortune** [comment]
Ah!  Thanks for that @0xmiroslav  - I see it now!

**Comment by aditygrg2** [comment]
Hello @slafortune What are your thoughts on my proposal [here](https://github.com/Expensify/App/issues/23422#issuecomment-1666864896)

**Comment by slafortune** [comment]
I'll defer to @0xmiroslav and @roryabraham for proposal reviews and selection 👍

**Comment by aditygrg2** [comment]
@0xmiroslav @roryabraham What are your views on my [proposal](https://github.com/Expensify/App/issues/23422#issuecomment-1666864896)?

**Comment by aditygrg2** [comment]
@0xmiroslav I did tried that solution, but it didn't worked for me.

**Comment by aditygrg2** [comment]
@ginsuma It works on Mac/Safari too right?

**Comment by ishpaul777** [comment]
i tried adding `willChange: "transform"` sometimes it does not work as expected and requires a reload to work as expected

https://github.com/Expensify/App/assets/104348397/23acd254-332a-47ec-8894-d9bfc8886655

**Comment by ishpaul777** [comment]
Steps to repro:
1. reload app while on other screen (without illustration)
2. navigate to preferences/workspaces.
3. Scroll up or down as soon as you can.(if wait more than 2-3 seconds, then works as expected).

**Comment by slafortune** [comment]
@0xmiroslav is there a proposal we can move forward with?

**Comment by roryabraham** [comment]
Sorry for the delay, fortunately this isn't a very high-priority issue so not a big deal!

**Comment by roryabraham** [comment]
Oh, sorry I had missed [this comment](https://github.com/Expensify/App/issues/23422#issuecomment-1697727842)

**Comment by roryabraham** [comment]
Seems like all the solutions that just move the overscroll spacer up in the view hierarchy create the same issue on the bottom, as shown in the videos from [this comment](https://github.com/Expensify/App/issues/23422#issuecomment-1666864896)

**Comment by roryabraham** [comment]
> we can play with onScroll Prop on ScrollView to dynamically set the background color. (this may result in performance regression)

Tried this, and it was janky

**Comment by roryabraham** [comment]
While not perfect, it seems like [@ishpaul777's proposal](https://github.com/Expensify/App/issues/23422#issuecomment-1666890994) might be the best we can do. In practice you're right that you'd have to be intentionally over-scrolling with two fingers in order to actually see this bug with this change, so I suppose that's better than what we have now.

I do think it's essentially a safari bug, but since that's out-of-scope for us to worry about fixing I suppose a workaround is the best we can do in this case.

**Comment by aditygrg2** [comment]
Hello @roryabraham,

I guess both proposals are exactly the same with just a change in height of the background. I am still fine with your review, but just wanted to check with you, if you are aware of this.

**Comment by ishpaul777** [comment]
@0xmiroslav @roryabraham PR up for review

**Comment by slafortune** [comment]
@0xmiroslav are you able to get to this review?

Reassigning a BZ member, I am out on sabbatical

**Comment by roryabraham** [comment]
PR reviewed and approved but intentionally not merged right now, see [this comment](https://github.com/Expensify/App/pull/26873#issuecomment-1712894564)

**Comment by abekkala** [comment]
waiting on merge

**Comment by abekkala** [comment]
@roryabraham would a bonus be added here based on the PR being intentionally not merged?

**Comment by abekkala** [comment]
@0xmiroslav can you complete the checklist above?

**Comment by abekkala** [comment]
### PAYMENTS TO BE MADE:

- @narzullome (Reporter) [$250] (reported before price change)

- @ishpaul777 (Contributor) [$1000 + 50% bonus]
- @0xmiroslav (Reviewer) [$1000 + 50% bonus]

**Comment by abekkala** [comment]
@ishpaul777 payment sent and contract ended - thank you! 🎉 

@narzullome I've sent you the $250 reporting payout offer, once accepted I can initiate payment - thanks!

**Comment by ishpaul777** [comment]
Payment received! Thank you 🎉

**Comment by narzullome** [comment]
@abekkala Offer accepted

**Comment by abekkala** [comment]
waiting on payment for 

@0xmiroslav (Reviewer) [$1000 + 50% bonus]

on hold until we determine how payments are to be made

**Comment by abekkala** [comment]
@0xmiroslav are you ready to accept payments via Upwork yet?
